tests/unit/test_generate_umm_var.py

```diff
@@ -0,0 +1,276 @@
+        mock_granule_query.return_value.parameters.assert_called_once_with(
+            downloadable=True,
+            sort_key='-start_date',
+            concept_id=self.collection_concept_id
+        )
+
+        # Ensure the call to download the granule had correct parameters
+        mock_download_granule.assert_called_once_with(self.netcdf4_url,
+                                                      self.bearer_token_header,
+                                                      out_directory=ANY)
+
+        # Ensure the output looks as expected
+        self.assertSetEqual(set(published_umm_var),
+                            set(expected_concept_ids))
+
+    @patch('varinfo.umm_var.publish_umm_var')
+    @patch('varinfo.cmr_search.GranuleQuery')
+    @patch('varinfo.generate_umm_var.download_granule')
+    def test_error_from_search_is_raised(self, mock_download_granule,
+                                         mock_granule_query,
+                                         mock_publish_umm_var):
+        """ Ensure an error raised iduring search is propagated out to the
+            user.
+
+        """
+        # Simulate no results found to trigger an IndexError
+        mock_granule_query.return_value.get.return_value = []
+
+        # Run the test, ensuring expected exception is raised:
+        with self.assertRaises(IndexError) as context_manager:
+            generate_collection_umm_var(self.collection_concept_id,
+                                        self.bearer_token_header)
+
+        self.assertEqual(str(context_manager.exception),
+                         'No granules were found with selected parameters '
+                         'and user permissions')
+
+        # Ensure the granule query used expected query parameters
+        mock_granule_query.return_value.parameters.assert_called_once_with(
+            downloadable=True,
+            sort_key='-start_date',
+            concept_id=self.collection_concept_id
+        )
+
+        # Ensure no attempt was made to download the granule after failure:
+        mock_download_granule.assert_not_called()
+
+        # Ensure no attempt was made to publish UMM-Var after the failure:
+        mock_publish_umm_var.assert_not_called()
+
+    @patch('varinfo.umm_var.publish_umm_var')
+    @patch('varinfo.cmr_search.GranuleQuery')
+    @patch('varinfo.generate_umm_var.download_granule')
+    def test_publishing_errors(self, mock_download_granule, mock_granule_query,
+                               mock_publish_umm_var):
+        """ Show the output list from publication will correctly handle a
+            response with an error.
+
+        """
+        error_message = 'Invalid JSON'
+        concept_ids = [
+            'V0000000001-PROV', 'V0000000002-PROV', 'V0000000003-PROV',
+            'V0000000004-PROV', 'V0000000005-PROV', 'V0000000006-PROV',
+            'V0000000007-PROV'
+        ]
+        concept_ids_and_error = concept_ids + [error_message]
```

Comment:
I am having a hard time visualizing what this will look like can you either add a `# e.g. ...` or fully declare `concept_ids_and_error` with `error_message` embedded?